cmd/config.go

```diff
@@ -86,6 +95,24 @@ func (d *DBConfig) URL() (string, error) {
 	return d.DBConnect, nil
 }
 
+// DefaultDBConfig is a temporary helper function that copies DBConfig
+// fields unmarshalled from the root of a component's JSON config
+// (deprecated) to the named `DBConfig` substruct of the service config.
+// TODO(#5275): Remove once all configs in dev, staging and prod
+// have been updated to contain the `dbconfig` field
+func DefaultDBConfig(dbConfig *DBConfig, databaseConfig *DatabaseConfig) {
+	if databaseConfig.DBConnectFile == "" {
```

Comment:
I think this should test for `if dbConfig.DBConnectFile != ""`. As is, this will cause the old embedded version to overwrite the new nested version if both are present. If both are present, we'd rather prefer the new nested version.